collector/netframework_clrmemory.go

```diff
@@ -0,0 +1,349 @@
+			prometheus.BuildFQName(Namespace, subsystem, "number_gc_handles"),
+			"Displays the current number of garbage collection handles in use. Garbage collection handles are handles to resources external to the common language runtime and the managed environment.",
+			[]string{"process"},
+			nil,
+		),
+		NumberGen0Collections: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "gen0_collections_total"),
+			"Displays the number of times the generation 0 objects (that is, the youngest, most recently allocated objects) are garbage collected since the application started.",
+			[]string{"process"},
+			nil,
+		),
+		NumberGen1Collections: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "gen1_collections_total"),
+			"Displays the number of times the generation 1 objects are garbage collected since the application started.",
+			[]string{"process"},
+			nil,
+		),
+		NumberGen2Collections: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "gen2_collections_total"),
+			"Displays the number of times the generation 2 objects are garbage collected since the application started.",
+			[]string{"process"},
+			nil,
+		),
+		NumberInducedGC: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "induced_gc_total"),
+			"Displays the peak number of times garbage collection was performed because of an explicit call to GC.Collect.",
+			[]string{"process"},
+			nil,
+		),
+		NumberofPinnedObjects: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "number_pinned_objects"),
+			"Displays the number of pinned objects encountered in the last garbage collection.",
+			[]string{"process"},
+			nil,
+		),
+		NumberofSinkBlocksinuse: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "number_sink_blocksinuse"),
+			"Displays the current number of synchronization blocks in use. Synchronization blocks are per-object data structures allocated for storing synchronization information. They hold weak references to managed objects and must be scanned by the garbage collector.",
+			[]string{"process"},
+			nil,
+		),
+		NumberTotalCommittedBytes: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "committed_bytes"),
+			"Displays the amount of virtual memory, in bytes, currently committed by the garbage collector. Committed memory is the physical memory for which space has been reserved in the disk paging file.",
+			[]string{"process"},
+			nil,
+		),
+		NumberTotalreservedBytes: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "reserved_bytes"),
+			"Displays the amount of virtual memory, in bytes, currently reserved by the garbage collector. Reserved memory is the virtual memory space reserved for the application when no disk or main memory pages have been used.",
+			[]string{"process"},
+			nil,
+		),
+		TimeinGC: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "gc_timein"),
```

Comment:
Like the JIT counter, `gc_time_seconds_total`

Reply:
Yes but this one seems to be the time of the last GC and not the total time spent doing GC so I would remove the "total" suffix. I need to double check.

Reply:
It is not monotonically increasing, so definitely a gauge, the value is > 100 so not a percentage and too high to be seconds. It could be microseconds or simply a number of cycles.